functions/recursion.js

```diff
@@ -0,0 +1,27 @@
+module.exports = function recursion(tree) {
+    let count = 0;
+    let tempArr = [];
+    const arrOfTree = [];
+    const tempResult = [];
+    arrOfTree.push(tree);
+    const result = function repeat() {
+        const currentEl = arrOfTree.shift();
+        tempArr.push(currentEl.value);
+        if (currentEl.left) {
+            arrOfTree.push(currentEl.left);
+        }
+        if (currentEl.right) {
+            arrOfTree.push(currentEl.right);
+        }
+        if (tempArr.length === Math.pow(2, count)) {
+            tempResult.push(tempArr);
+            tempArr = [];
+            count += 1;
+        }
+        for (let i = 0; i < arrOfTree.length; i += 1) {
+            repeat(arrOfTree[i]);
```

Comment:
` const result = function repeat()`
looks like function repeat don't have any arguments =), why we call it with `arrOfTree[i]`